flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/legacy/messages/DashboardConfiguration.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.rest.handler.legacy.messages;
+
+import org.apache.flink.runtime.rest.handler.legacy.DashboardConfigHandler;
+import org.apache.flink.runtime.rest.messages.ResponseBody;
+import org.apache.flink.runtime.util.EnvironmentInformation;
+import org.apache.flink.util.Preconditions;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.time.ZonedDateTime;
+import java.time.format.TextStyle;
+import java.util.Locale;
+
+/**
+ * Response of the {@link DashboardConfigHandler} containing general configuration
+ * values such as the time zone and the refresh interval.
+ */
+public class DashboardConfiguration implements ResponseBody {
+
+	public static final String FIELD_NAME_REFRESH_INTERVAL = "refresh-interval";
+	public static final String FIELD_NAME_TIMEZONE_OFFSET = "timezone-offset";
+	public static final String FIELD_NAME_TIMEZONE_NAME = "timezone-name";
+	public static final String FIELD_NAME_FLINK_VERSION = "flink-version";
+	public static final String FIELD_NAME_FLINK_REVISION = "flink-revision";
+
+	@JsonProperty(FIELD_NAME_REFRESH_INTERVAL)
+	private final long refreshInterval;
+
+	@JsonProperty(FIELD_NAME_TIMEZONE_NAME)
+	private final String timeZoneName;
+
+	@JsonProperty(FIELD_NAME_TIMEZONE_OFFSET)
+	private final int timeZoneOffset;
+
+	@JsonProperty(FIELD_NAME_FLINK_VERSION)
+	private final String flinkVersion;
+
+	private final String flinkRevision;
```

Comment:
this field is missing the Jackson annotation.

Reply:
Which means we would want a Json marshalling test for this also.

Reply:
Good catch. I'll add the marshalling test.